### ISSUE TITLE
Add some LaTeX templates and ability to use them in Open LaTeX Studio

### ISSUE THREAD

**Issue by sebbrudzinski** [sebbrudzinski/Open-LaTeX-Studio#27]
As a part of this ticket we should first create some sample LaTeX templates - for example for articles, letters, books, CVs and add them to the code repository. We should then allow to load such templates in the Open LaTeX Studio. This should be exposed as an option in the File menu (perhaps named Open template).

**Comment by aQaTL** [comment]
As my first task, I think this one will be ok.

**Comment by aQaTL** [comment]
For now, I have already created this:
![templatedialog](https://cloud.githubusercontent.com/assets/17130832/14437352/1a836346-0021-11e6-954d-6bf9d33fc12a.png)
You can find code and commits on my GitHub: [aQaTL's Open-LaTeX-Studio fork](https://github.com/aQaTL/Open-LaTeX-Studio)

How it works: firstly program checks if JTextArea (this one in second box) contains correct path to file - if yes, then loads it, if no - loads selected template from list (if you choose anything in list, it clears JTextArea). So JTextArea literally overrides selection in JList. 

Unfortunately, I haven't  added any templates yet. I don't know, how to make them automatically added to Templates folder (I created it manually in app directory) in installation process. 

It's my first commit, so any feedback will be really helpful for me.

**Comment by sebbrudzinski** [comment]
@aQaTL What exactly would the "Choose your own template" do? Concerning loading of the templates - first of all, we create multiple files under the application directory. You can take a look how it's done in the code. On the other hand, I'm not sure why would you want to place the templates in the app directory. I think they could be simply placed in the resources package in the codebase (unless you plan to make the templates selection extendible by the end users, but that is a bigger task).

**Comment by aQaTL** [comment]
@sebbrudzinski I put templates in app directory, just to test if its working (and I thought that it's correct place), but I admit, that placing it in resources is smarter (I will move it). I am not planning to do templates section extendible - for that, I created "Choose your own template" section. So, this section would allow user to use his own LaTeX templates files (if it's doesn't make sense I can delete it).

**Comment by sebbrudzinski** [comment]
@aQaTL Correct me if I'm wrong, but I'm thinking that it would do exactly the same thing as "Open File", which we already have under the File menu. If that's the case, we should probably drop it.

**Comment by aQaTL** [comment]
@sebbrudzinski Oh, That's true. I didn't think of it that way. Thanks for explaining. How about the rest of code (isn't it bad, too complicated, or something like that)?

**Comment by sebbrudzinski** [comment]
@aQaTL I've taken a brief look only and it's not bad. One thing that caught my eye is the placement of braces. As per java and our project coding convention, the opening brace should be in the same line as the instruction (if/method/class etc.). Moreover, it would probably be better to design the panel using the NetBeans GUI designer, so other people can easily adjust it at a later point. I'll do a complete review once a pull request is opened.

**Comment by aQaTL** [comment]
@sebbrudzinski Braces placement aren't a problem, I can switch it in NetBeans options, press alt + shift + f and they will match to project convention (I forgot to do that before committing). 
Today I realized that everything here is designed with the NetBeans GUI builder and it's a good idea to use it instead of hard-coding entire look. I think it's the easy part of this task. Once I have moved the templates folder to resources it's not easy to operate on them (I assume they will be stored in a jar file). Additionally, I am still learning about LaTeX, so I haven't started working on templates yet.

**Comment by aQaTL** [comment]
@sebbrudzinski There's a big problem with operating on files inside the jar - everything in OpenLaTeXStudio uses File object and I can't create a File object from a jar resources(I tried many options). So, even if I load file, I can't pass anything to setCurrentFile method and NullPointerException occurs. I spent a lot of time thinking about it and.. How about placing these files in a jar and unpacking it to normal directory at first run(in an app directory for example)? 
One more thing - what do you think about including some templates from this website [LaTeX templates](http://www.latextemplates.com)? I contacted the owner and I have his permission to use(I can show you it). Just need to mark his website as a source. In my opinion a link in "Open Template" Dialog will be enough and fair. Note that these templates will be much better than mine.

**Comment by sebbrudzinski** [comment]
@aQaTL I think it's a wonderful idea to use templates from latextemplates.com  Thanks for contacting them about it. I've created a quick mockup, how selecting a template could look like. What do you think about it? I think it may make sense to specify source per single template, in case we use more templates from other websites in the future.
[Load_template_mockup.pdf](https://github.com/sebbrudzinski/Open-LaTeX-Studio/files/221711/Load_template_mockup.pdf)
Concerning jar resources - we do not really need to convert them to File objects or invoke "setCurrentFile" method. All we should do is extract the content of the file and display it in the Editor window. You can use the loading of the welcome.tex as an inspiration.

**Comment by aQaTL** [comment]
@sebbrudzinski I am glad that you like my idea and I think that your vision makes sense.  Take a look at it:
![open_template_feature_implementation](https://cloud.githubusercontent.com/assets/17130832/14582593/070e5d24-040a-11e6-8872-aaf0c3b864e2.png)
I decided to specify information about templates in templates.xml file (stored in resources), so if somebody will want to add new template, just need to pase it in the templates folder and add a few lines to xml file. 
Tell me, if the code is too complicated (I had to learn xml write it). 
I've already pushed changes to my repository.
